pkg/kube/object_patch/spec_generator.go

```diff
@@ -0,0 +1,156 @@
+package object_patch
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type specGenerator struct {
```

Comment:
I think "SpecGenerator" is confusing. It is more a `PatchCollector` — similar to the `MetricCollector`.

Reply:
Indeed. It was something like `DeferredObjectPatcher` but PatchCollector look great